### PR TITLE
Retry failed connection

### PR DESCRIPTION
**Pull request by benSlaughter**
* Added retry failed connection for requests  
    When a request gets a failed connection error then the request will be retried.
    This is to catch the Faraday::ConnectionFailed that has been seen recently.
    I have included warnings so as to try to ensure that it does not go unnoticed.
* Added connection attempts for connection retries
    The user can now configure the number of connection attempts.
    The default has been set to 1 connection attempt.
    This will be no retries.
    A retry will happen if the tries is lower than the configured connection_attempts
* Fixed minor Rubocop error in configuration

### CODE REVIEW

File: lib/bwapi/request.rb [hunk 1]
```diff
@@ -57,6 +57,23 @@ def patch(path, opts = {})
     # @param opts [Hash] Request parameters
     # @return [Hash] Response
     def request(method, path, opts = {})
+      retries ||= 0
+      send_request(method, path, opts)
+    rescue Faraday::ConnectionFailed => e
+      warn "Faraday failed to connect with the error: #{e.message}"
+      raise e unless (retries += 1) < 3
```
Comment (spmason) [inline lib/bwapi/request.rb]: Maybe this should be configurable, with 0 retries the default?

File: lib/bwapi/request.rb [hunk 2]
```diff
@@ -57,6 +57,23 @@ def patch(path, opts = {})
...
+      warn "Reseting and retrying. Retry: #{retries}"
```
Comment (spmason) [inline lib/bwapi/request.rb]: Typo: `resetting`

**Comment by benSlaughter** [comment]
@spmason updated, ready for review

**Comment by benSlaughter** [comment]
test this please

**Comment by benSlaughter** [comment]
test this please

**Comment by benSlaughter** [comment]
test this please

**Pull request merged**